common/test/src/ui/tst_MapDocument.cpp

```diff
@@ -462,6 +462,46 @@ TEST_CASE_METHOD(MapDocumentTest, "canUpdateLinkedGroups")
     document->canUpdateLinkedGroups(kdl::vec_static_cast<mdl::Node*>(entityNodes)));
 }
 
+TEST_CASE_METHOD(MapDocumentTest, "newDocument")
+{
+  SECTION("Default worldspawn properties")
```

Comment:
It would be good to also cover the case where the default properties should not be set. It is easier to cover such tests with a `GENERATOR` and a table of input mapped to expected output. There are plenty of examples in the test code. In this case it would look something like this:

```
using T = std::tuple<bool, std::vector<mdl::EntityProperty>>;

// clang-format off
const auto& 
[setDefaultProperties, expectedEntityProperties] = GENERATE(values<T>({
{false,                           {{mdl::EntityPropertyKeys::Classname, mdl::EntityPropertyValues::WorldspawnClassname}}},
{true,                            {{mdl::EntityPropertyKeys::Classname, mdl::EntityPropertyValues::WorldspawnClassname},
                                      {"some_default_prop", "value"}}},
}));
// clang-format on

...

CHECK_THAT(
      worldNode->entity().properties(),
      Catch::Matchers::UnorderedEquals(expectedEntityProperties));
```

Reply:
Yeah, before looking into that, I would like to know if loading entity definitions from a mock file is correct.